### PR TITLE
Documented tf.keras.preprocessing purpose

### LINKED ISSUE #37445: Document purpose for tf.keras.preprocessing module
## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing

## Description of issue (what needs changing):

The module documentation is very terse and only states that it contains "preprocessing utils". It does not state the specific purpose.

It would be helpful if it defined the intended use of the `tf.keras.preprocessing` module (_e.g._ to clean up or transform tf.data.Datasets before they are fed to the model).

Also, since the `tf.feature_column` module has similar functionality, it would be nice to describe when to use one or the other, or how they are intended to be used together.

### PR DESCRIPTION
**Pull request by ManishAradwad**
Fixes : #37445 
Docs about `tf.feature_column` not added yet since I'm not sure about the differences between them. Can someone plz help about that??

### CODE REVIEW

File: tensorflow/python/keras/preprocessing/__init__.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
-"""Keras data preprocessing utils."""
+"""Provides keras data preprocessing utils to pre-process tf.data.Datasets before they are fed to the model."""
```
Comment (tanzhenyu) [inline tensorflow/python/keras/preprocessing/__init__.py]: Is this intended to be used with tf.data.Dataset? IIUC it should be the layers under tensorflow/python/keras/layers/preprocessing

Comment (ManishAradwad) [reply tensorflow/python/keras/preprocessing/__init__.py]:
```suggestion
"""Provides keras data preprocessing utils to pre-process data fed to the keras layers."""
```
Will this be okay?

Comment (gbaned) [reply tensorflow/python/keras/preprocessing/__init__.py]: @tanzhenyu Can you please take a look on the above comment from @ManishAradwad. Thank you!

Comment (gbaned) [reply tensorflow/python/keras/preprocessing/__init__.py:15]: @tanzhenyu Any update on this PR? Please. Thanks!

Comment (gbaned) [reply tensorflow/python/keras/preprocessing/__init__.py:15]: @tanzhenyu Any update on this PR? Please. Thanks!

**Review by tanzhenyu — APPROVED** [review]

**Pull request merged**